assets/aml-benchmark/components/src/prompt_crafter/package/prompt_factory.py

```diff
@@ -190,6 +191,9 @@ def process_row(self, row: Dict, index: int) -> Dict:
         if self.output_pattern is not None:
             output_data['completion'] = self.get_label_from_output_pattern(row)
 
+        if self.ground_truth_column_name is not None and self.ground_truth_column_name in row:
```

Comment:
If `self.ground_truth_column_name` is not None and not an empty string, we should check `self.ground_truth_column_name` should exist in `row`. If not, we should throw an error saying the column does not exist in the data.